lib/singlepass-backend/src/machine.rs

```diff
@@ -83,7 +83,14 @@ impl Machine {
 
     /// Releases a temporary GPR.
     pub fn release_temp_gpr(&mut self, gpr: GPR) {
-        assert_eq!(self.used_gprs.remove(&gpr), true);
+        assert!(self.used_gprs.remove(&gpr));
+    }
+
+    /// Specify that a given register is in use.
+    pub fn reserve_temp_gpr(&mut self, gpr: GPR) -> GPR {
```

Comment:
Could you make the name a bit more distinguishable to `acquire_temp_gpr`, like `reserve_unused_gpr` ?

Reply:
Done. I wanted to keep `temp` part of the name so that it refers to the RAX, RCX, RDX list of registers, not the RSI ... R11 list.

Reply:
Okay, I thought this was used to reserve the RSI..R11 registers :)

Reply:
Might be worth adding a comment explaining this if the new name doesn't make this explicit.  As someone without context on this, I'm sure it'd take be a bit to figure this out

Reply:
The new name is `reserve_unused_temp_gpr`. It also has a comment on the function, and an assertion in the function. I'm in the position where I know what it does, so I can't see what about it might be confusing. Let me know if I can improve it?